assert/partial.go

```diff
@@ -0,0 +1,79 @@
+package assert
+
+import (
+	"reflect"
+	"testing"
+)
+
+func (assert *Assert) Partial(obj interface{}, target interface{}) {
```

Comment:
shouldn't we support the same formatting/description mechanism here ? or else there is no way for users to provide their own customized error message when the partial assert fails

Reply:
also we need to call assert.t.Helper() here no ?

Reply:
Just saw we pass the details on the assert constructor, but I'm a little confused, some assert methods still get details as a param for the method: https://github.com/madlambda/spells/pull/21/files?diff=unified&w=0#diff-0e9d71fd6e2e96e944d107325b0a19d470828147aa4307147e40c45c09a08344R12

It doesn't seem that tha details passed on the constructor are used: https://github.com/madlambda/spells/pull/21/files?diff=unified&w=0#diff-7695f44167b7ec89b59ff2da3c86237a9997f9974e92a6b9691e90722ed46a2cR34

And here there are no details at all as a parameter, but then the details passed on the constructor doesn't seem to be used either.

Or maybe I'm missing something 🤔